spark/src/main/scala/org/apache/mahout/sparkbindings/drm/CheckpointedDrmSpark.scala

```diff
@@ -78,8 +78,8 @@ class CheckpointedDrmSpark[K: ClassTag](
   }
 
   def cache() = {
-    if (!cached) {
-      rdd.persist(_cacheStorageLevel)
+    if (!cached && _cacheStorageLevel != StorageLevel.NONE) {
```

Comment:
NONE means yes, it's not cached.

The material effect of setting "NONE" cache level in spark, as far as i
understand, is exactly as not setting anything at all; the difference is
once you set it to anything, you cannot change it. So this means if i call
"NONE" it would lock me to that choice with spark (which did not make much
sense to me) otherwise it doesn't matter.

On Thu, Jun 11, 2015 at 3:56 PM, Andrew Musselman <notifications@github.com>
wrote:

> In
> spark/src/main/scala/org/apache/mahout/sparkbindings/drm/CheckpointedDrmSpark.scala
> <https://github.com/apache/mahout/pull/135#discussion_r32277244>:
>
> > @@ -78,8 +78,8 @@ class CheckpointedDrmSpark[K: ClassTag](
> >    }
> >
> >    def cache() = {
> > -    if (!cached) {
> > -      rdd.persist(_cacheStorageLevel)
> > +    if (!cached && _cacheStorageLevel != StorageLevel.NONE) {
>
> Will this be confusing if someone tries to call cache() with NONE set and
> it's not cached?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/mahout/pull/135/files#r32277244>.
>